Honeycomb Bravo.lua

```diff
@@ -116,396 +133,617 @@ function array_has_true(array)
+		local bank = led_position[1]
+		local bit = led_position[2]
 
-	local master_state = false
-	local buffer_modified = false
-
-	function get_led(led)
-		return buffer[led[1]][led[2]]
+		-- Check if the corresponding bit in data is on (true)
+		if buffer[bank] and buffer[bank][bit] == true then
+			table.insert(on_leds, led_name)
+		end
 	end
 
-	function set_led(led, state)
-		if state ~= get_led(led) then
-			buffer[led[1]][led[2]] = state
-			buffer_modified = true
+	if #on_leds > 0 then
+		local on_leds_string = table.concat(on_leds, ", ")
+		if on_leds_string == last_buffer_str then
+			return
 		end
+		last_buffer_str = on_leds_string
+		write_log("LEDs ON: " .. on_leds_string) -- Using print as a stand-in for your actual logging mechanism
 	end
+end
 
-	function all_leds_off()
-		for bank = 1, 4 do
-			buffer[bank] = {}
-			for bit = 1, 8 do
-				buffer[bank][bit] = false
-			end
-		end
+-- Support variables & functions for sending LED data via HID
+
+local buffer = {}
 
+local master_state = false
+local buffer_modified = false
+
+function get_led(led)
+	return buffer[led[1]][led[2]]
+end
+
+function set_led(led, state)
+	if state ~= get_led(led) then
+		buffer[led[1]][led[2]] = state
 		buffer_modified = true
 	end
+end
 
-	function send_hid_data()
-		local data = {}
+function all_leds_off()
+	for bank = 1, 4 do
+		buffer[bank] = {}
+		for bit = 1, 8 do
+			buffer[bank][bit] = false
+		end
+	end
 
-		for bank = 1, 4 do
-			data[bank] = 0
+	buffer_modified = true
+end
 
-			for bit = 1, 8 do
-				if buffer[bank][bit] == true then
-					data[bank] = bitwise.bor(data[bank], bitwise.lshift(1, bit - 1))
-				end
-			end
-		end
+function send_hid_data()
+	local data = {}
 
-		local bytes_written = hid_send_filled_feature_report(bravo, 0, 65, data[1], data[2], data[3], data[4]) -- 65 = 1 byte (report ID) + 64 bytes (data)
+	for bank = 1, 4 do
+		data[bank] = 0
 
-		if bytes_written == -1 then
-			write_log('ERROR Feature report write failed, an error occurred')
-		elseif bytes_written < 65 then
-			write_log('ERROR Feature report write failed, only '..bytes_written..' bytes written')
-		else
-			buffer_modified = false
+		for bit = 1, 8 do
+			if buffer[bank][bit] == true then
+				data[bank] = bitwise.bor(data[bank], bitwise.lshift(1, bit - 1))
+			end
 		end
 	end
-
-	-- Initialize our default state
-	all_leds_off()
-	send_hid_data()
-	hid_open(10571, 6401) -- MacOS Bravo must be reopened for .joy axes to operate
-
-	-- Change LEDs when their underlying dataref changes
-	-- Bus voltage as a master LED switch
-	local bus_voltage = dataref_table('sim/cockpit2/electrical/bus_volts')
-	if PROFILE == "laminar/CitX" then
-		-- On the C750 bus_volts is only on when the Standby Power is on, but the lights and indicators in the sim are on before that
-		bus_voltage = dataref_table('laminar/CitX/APU/DC_volts')
-	elseif PROFILE == "B738" then
-		bus_voltage = dataref_table('laminar/B738/electric/batbus_status')
+	local bytes_written = 2147483647
```

Comment:
This is a little enigmatic I think. Perhaps we can just set it to something in the `else` branch rather than initialising it here?